[WIP]849 send notification when broadcast is created

faithngetich opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

#### Description
<!--- Describe your changes in detail and link to the issue. eg: Fixes -->
 Fixes #849
#### Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
It would be nice to send an email/slack notification informing that a new broadcast has been created.

#### How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
You will need to login to the site and create a broadcast, an easy way of doing so is as follows:
- Go to this [link](http://localhost:4200/find-broadcasts) and search for a broadcast that does not exist. 
- You'll be redirected to a page where you can create a broadcast. When you fill in the details and hit the create button you will be able to trigger the email. 
- Open mailer should send you to a link with the email details.
-  If you're not redirected, relax you can still view the emails on this [link](http://localhost:8025). This will show you all the emails that you have attempted sending. 

#### Screenshots (if appropriate):
When you have successfully created a broadcast
<img width="1440" alt="screen shot 2018-10-25 at 19 26 00" src="https://user-images.githubusercontent.com/23743836/47515582-ee029300-d88b-11e8-9bb2-87e7d3981da8.png">
Just before you save the broadcast
<img width="1440" alt="screen shot 2018-10-25 at 19 25 42" src="https://user-images.githubusercontent.com/23743836/47515608-feb30900-d88b-11e8-91c8-91c0162440d0.png">

#### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes. -->
- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

faithngetich committed:
send emails on broadcast notification

faithngetich committed:
Add letter opener to view sent mail locally

faithngetich committed:
Remove commented code

faithngetich committed:
Add email notification on broadcast creation job and test it

faithngetich committed:
Test email notification on broadcast creation mailing method

faithngetich committed:
Add email template

roschaefer committed:
Remove letter opener

roschaefer committed:
Add service mailhog to catch mails in development

..and configure sidekiq to process mailer queue

roschaefer committed:
Sketch UserMailer#ask_for_spam_check and test

roschaefer committed:
Make sure we have moderators in the future

roschaefer committed:
Add documentation for mailhog

roschaefer committed:
Remove obsolete code

roschaefer committed:
Remove obsolete mailer view

roschaefer committed:
Turn the user into a moderator only on requests

roschaefer commented:
@faithngetich I updated your PR quite a bit and I hope I paved the way for you with some tests. I replaced `letter_opener` with `mailhog` which is the cleaner alternative, as it does not require a change to our rails code. Also, I configured `sidekiq` to process our mailers.

roschaefer committed:
Remove unnecessary code

roschaefer committed:
Fix rspec test - test inside context block

NickSchimek committed:
Add mail body

faithngetich committed:
test that email body provides a deep link to check the broadcast

faithngetich committed:
Add [rundfunk mitbestimmen] to the email subject

NickSchimek committed:
Add internationalization to email subject

ciremoussadia committed:
Fixes the test that checks that notification are sent to all admins and moderators at Sidekiq queue list.

ciremoussadia committed:
Turns anyone who create or update a broadcast to a moderator

This lead to an error on specs on which email notification were sent to admins and moderator. We fix it by adding 1 to the size total consideing that the creator of the broadcats is no turned to a moderator.

ciremoussadia commented:
Hello @roschaefer, what can I do to help finish this PR?

roschaefer commented:
@ciremoussadia sorry for responding so late. I kept the Github notification "unread" to motivate me to dig into this PR but I failed.

So, to answer your question: The build server tells me that `push` has succeeded but `pr` has not. This could mean that branch `master` was updated and has not been merged into this PR.

1. `git checkout `849_send_notification_when_broadcast_is_created`
2. `git fetch`
3. `git merge origin/master`
4. `git push origin -u 849_send_notification_when_broadcast_is_created`

Will update this PR. The expected outcome is that both `pr` and `push` fail. The build server tells me that there are rubocop errors (easy) and some rspec failures. You could fix the rubocops with `bundle exec rubocop -a`. As a next step you would need to fix the rspec tests. I think once this is done we could merge the PR.

ciremoussadia committed:
Merge branch 'master' into 849_send_notification_when_broadcast_is_created

ciremoussadia committed:
Fixes rubocop offenses.
